.github/workflows/docker_check.yml

```diff
@@ -39,10 +67,86 @@ jobs:
         uses: actions/checkout@v3
 
       - name: Build all platforms
-        uses: docker/build-push-action@v2
+        uses: docker/build-push-action@v6
         with:
           context: .
           push: false
+          load: true
+          tags: "localhost/sygnal:latest"
           labels: "gitsha1=${{ github.sha }}"
           file: "docker/Dockerfile"
           platforms: linux/amd64,linux/arm64
+
+          cache-from: type=gha
+          cache-to: type=gha,mode=max
+
+      - name: Save container image to disk
+        run: |
+          docker image save localhost/sygnal:latest --output ${{ runner.temp }}/sygnal_image.tar
+
+      # https://docs.docker.com/build/ci/github-actions/share-image-jobs/
+      - name: Upload container image for subsequent steps
+        uses: actions/upload-artifact@v4
+        with:
+          name: sygnal_image
+          path: ${{ runner.temp }}/sygnal_image.tar
+          retention-days: 1
+
+  proxytest:
+    name: Check Proxy functionality
+    runs-on: ubuntu-latest
+    needs: build
+    steps:
+      - name: Download container image from build step
+        uses: actions/download-artifact@v4
+        with:
+          name: sygnal_image
+          path: ${{ runner.temp }}
+
+      - name: Load image into container engine
+        run: |
+          docker image load --input ${{ runner.temp }}/sygnal_image.tar
+
+      - uses: actions/checkout@v4.2.2
+
+      - name: Prepare test setup
+        run: |
+          scripts-dev/proxy-test/setup.sh
+          # Chown the mitmproxy setup files to be group-owned by root,
+          # because mitmproxy will re-number its user and group at startup,
+          # but the group with that ID must already exist inside the container...
+          # See: https://github.com/mitmproxy/mitmproxy/issues/6840
+          sudo chown -R :root scripts-dev/proxy-test/mitmproxy
+
+      - name: Start test services in Compose file
+        uses: hoverkraft-tech/compose-action@8be2d741e891ac9b8ac20825e6f3904149599925 # v2.2.0
+        with:
+          compose-file: scripts-dev/proxy-test/docker-compose.yml
+
+      - name: Invoke test script
+        run: |
+          echo '====='
+          docker ps -a
+          # TODO do something less hacky than a static 5-second sleep
+          sleep 5
+          echo '====='
+          docker ps -a
+          echo '====='
+          docker exec sygnal sh /curl.sh notification-gcm.json
```

Comment:
Is this just waiting for sygnal to be reachable?
Should we be adding a `/health` endpoint to Sygnal and be relying on docker to wait for that?

Reply:
Good point. Have opened https://github.com/matrix-org/sygnal/pull/418 to add this.

Reply:
although not entirely sure if the compose action waits for it to be healthy :SS

Reply:
I added a loop that polls the healthiness (using healthcheck from https://github.com/matrix-org/sygnal/pull/418).